.github/workflows/ci.yml

```diff
@@ -0,0 +1,98 @@
+# TODO(#158): to be replaced with real service integration test.
+name: 'ci'
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - 'main'
+  workflow_dispatch:
+  workflow_call:
+
+env:
+  WIF_PROVIDER: 'projects/1096923323432/locations/global/workloadIdentityPools/github-pool-fd98/providers/github-provider'
+  WIF_SERVICE_ACCOUNT: 'jvs-fd98-ci-sa@jvs-ci-test.iam.gserviceaccount.com'
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}-test-unit'
+  cancel-in-progress: true
+
+jobs:
+  # Linting jobs - terraform, go, java
+  terraform_lint:
+    uses: 'abcxyz/pkg/.github/workflows/terraform-lint.yml@main' # ratchet:exclude
+    with:
+      directory: 'terraform'
+      terraform_version: '1.2'
+
+  go_lint:
+    uses: 'abcxyz/pkg/.github/workflows/go-lint.yml@main' # ratchet:exclude
+    with:
+      go_version: '1.20'
+
+  java_lint:
+    uses: 'abcxyz/pkg/.github/workflows/java-lint.yml@main' # ratchet:exclude
+    with:
+      java_version: '11'
+
+  # Unit tests - go, java
+  go_test:
+    uses: 'abcxyz/pkg/.github/workflows/go-test.yml@main' # ratchet:exclude
+    with:
+      go_version: '1.20'
+
+  java_test:
+    runs-on: 'ubuntu-latest'
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+      - uses: 'actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3' # ratchet:actions/checkout@v3
+      # Technically we don't need this step since we don't need anything from Artifact Registry.
+      # But the Artifact Registry wagon will keep retrying the authentication and blocking
+      # the unit test for a long time (likely a bug). As a result, we add this step to make
+      # the wagon happy.
+      - name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@ef5d53e30bbcd8d0836f4288f5e50ff3e086997d' # ratchet:google-github-actions/auth@v1
+        with:
+          workload_identity_provider: '${{ env.WIF_PROVIDER }}'
+          service_account: '${{ env.WIF_SERVICE_ACCOUNT }}'
+          # The Artifact Registry maven wagon looks for Google Application Default Credentials.
+          # https://github.com/GoogleCloudPlatform/artifact-registry-maven-tools
+      - name: Run test script
+        run: |-
+          mvn clean test --no-transfer-progress -f client-lib/java
+
+  integration:
```

Comment:
What's the plan for this? We should only have one integration test, right?

Reply:
Once we have real service integration test, I will remove this one entirely.

Reply:
Can you update the comment to make it clear?